api/certs/cert.go

```diff
@@ -112,19 +115,55 @@ func CertExists() (bool, bool, bool) {
 	if err != nil {
 		log.Error(err)
 	}
-	defaultDomain, err := util.GetDefaultDomainFromIP(config.Parameters.CertDomainName, config.Parameters.DefaultTlsDomainTmpl)
-	if err != nil {
-		log.Error(err)
+
+	for _, d := range config.Parameters.DefaultTlsDomainTmpl {
+		defaultDomain, err := util.GetDefaultDomainFromIP(config.Parameters.Hostname, d)
+		if err != nil {
+			log.Error(err)
+		}
+		_, _, certFile, keyFile := GetACMEFileNames(defaultDomain, config.Parameters.CertDirectory)
+		defaultCertExists, err = VerifyLocalCertificate(certFile, keyFile, defaultDomain)
+		if defaultCertExists {
+			break
+		}
 	}
-	_, _, certFile, keyFile := GetACMEFileNames(defaultDomain, config.Parameters.CertDirectory)
-	defaultCertExists, err := VerifyLocalCertificate(certFile, keyFile, defaultDomain)
-	if err != nil {
-		log.Error(err)
+
+	if len(config.Parameters.CertDomainName) > 0 && !defaultCertExists {
+		_, _, certFile, keyFile := GetACMEFileNames(config.Parameters.CertDomainName, config.Parameters.CertDirectory)
+		defaultCertExists, err = VerifyLocalCertificate(certFile, keyFile, config.Parameters.CertDomainName)
 	}
 
 	return wssUserCertExists, httpsUserCertExists, defaultCertExists
 }
 
+func getAvailableDomain() (string, error) {
+	for _, d := range config.Parameters.DefaultTlsDomainTmpl {
+		domain, err := util.GetDefaultDomainFromIP(config.Parameters.Hostname, d)
+		if err != nil {
+			return "", err
+		}
+		ok, err := checkDomain(domain)
+		if err != nil {
+			return "", err
+		}
+		if ok {
+			return d, nil
+		}
+	}
+	return "", errors.New("no domain available")
+}
+
+func checkDomain(domain string) (bool, error) {
```

Comment:
The check logic should be:
If certdomainname is an IP address, domain should resolve to that IP;
Otherwise, if hostname is an IP address, domain should resolve to that IP.